sdk/src/wallet/operations/syncing/addresses/output_ids/mod.rs

```diff
@@ -276,23 +276,24 @@ impl<S: 'static + SecretManage> Wallet<S> {
                 results = futures::future::try_join_all(tasks).await?;
             }
 
-            for res in results {
-                let (mut address, output_ids): (AddressWithUnspentOutputs, Vec<OutputId>) = res?;
+            let addresses_with_new_unspent_output_ids = results.into_iter().collect::<Result<Vec<_>, _>>()?;
+
+            for (mut address, new_unspent_output_ids) in addresses_with_new_unspent_output_ids {
                 // only return addresses with outputs
-                if !output_ids.is_empty() {
+                if !new_unspent_output_ids.is_empty() {
                     // outputs we had before, but now not anymore, got spent or are account/nft/foundries that don't
                     // get synced anymore because of other sync options
-                    for output_id in address.output_ids {
-                        if !output_ids.contains(&output_id) {
-                            spent_or_not_anymore_synced_outputs.push(output_id);
+                    for output_id in address.unspent_output_ids {
+                        if !new_unspent_output_ids.contains(&output_id) {
+                            spent_or_not_anymore_synced_outputs.push(output_id.into());
```

Comment:
Check clippy, this `.into()` isn't necessary

Reply:
remnant of me trying to make it a new-type instead of a type alias ... thanks

Reply:
Removed